mosipid-certify-integration-impl/src/main/java/io/mosip/certify/mosipid/integration/service/IdaVCIssuancePluginImpl.java

```diff
@@ -0,0 +1,216 @@
+			IdaVcExchangeRequest idaVciExchangeRequest = new IdaVcExchangeRequest();
+			CredentialDefinitionDTO vciCred = new CredentialDefinitionDTO();
+			idaVciExchangeRequest.setId(vciExchangeId);// Configuration
+			idaVciExchangeRequest.setVersion(vciExchangeVersion);// Configuration
+			idaVciExchangeRequest.setRequestTime(HelperService.getUTCDateTime());
+			idaVciExchangeRequest.setTransactionID(transaction.getAuthTransactionId());// Cache input
+			idaVciExchangeRequest.setVcAuthToken(transaction.getKycToken()); // Cache input
+			idaVciExchangeRequest.setIndividualId(individualId);
+			idaVciExchangeRequest.setCredSubjectId(holderId);
+			idaVciExchangeRequest.setVcFormat(vcRequestDto.getFormat());
+			idaVciExchangeRequest.setLocales(convertLangCodesToISO3LanguageCodes(transaction.getClaimsLocales()));
+			vciCred.setCredentialSubject(vcRequestDto.getCredentialSubject());
+			vciCred.setType(vcRequestDto.getType());
+			vciCred.setContext(vcRequestDto.getContext());
+			idaVciExchangeRequest.setCredentialsDefinition(vciCred);
+
+			String requestBody = objectMapper.writeValueAsString(idaVciExchangeRequest);
+			RequestEntity requestEntity = RequestEntity
+					.post(UriComponentsBuilder.fromUriString(vciExchangeUrl)
+							.pathSegment(transaction.getRelyingPartyId(),
+									identityDetails.get(CLIENT_ID).toString())
+							.build().toUri())
+					.contentType(MediaType.APPLICATION_JSON_UTF8)
+					.header(SIGNATURE_HEADER_NAME, helperService.getRequestSignature(requestBody))
+					.header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_NAME).body(requestBody);
+
+			ResponseEntity<IdaResponseWrapper<IdaVcExchangeResponse<JsonLDObject>>> responseEntity = restTemplate.exchange(
+					requestEntity, new ParameterizedTypeReference<IdaResponseWrapper<IdaVcExchangeResponse<JsonLDObject>>>() {});
+			if (responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
+				IdaResponseWrapper<IdaVcExchangeResponse<JsonLDObject>> responseWrapper = responseEntity.getBody();
+				if (responseWrapper != null && responseWrapper.getResponse() != null)
+				{
+					VCResult vCResult = new VCResult();
+					vCResult.setCredential(responseWrapper.getResponse().getVerifiableCredentials());
+					vCResult.setFormat(vcRequestDto.getFormat());
+					return vCResult;
+				}
+				log.error("Errors in response received from IDA VCI Exchange: {}", responseWrapper.getErrors());  //NOSONAR responseWrapper is already evaluated to be not null
+				throw new VCIExchangeException(CollectionUtils.isEmpty(responseWrapper.getErrors()) ?
+						ErrorConstants.VCI_EXCHANGE_FAILED : responseWrapper.getErrors().get(0).getErrorCode());
+			}
+			log.error("Error response received from IDA (VCI-exchange) with status : {}", responseEntity.getStatusCode());
+		} catch (VCIExchangeException e) { throw e; } catch (Exception e) {
+			log.error("IDA Vci-exchange failed ", e);
+		}
+		throw new VCIExchangeException();
```

Comment:
why are we thowing exception with no error message ?

Reply:
this is caught in certify service and a generic message of `vci-exchange-failed` is thrown from serice impl